node/src/components/storage.rs

```diff
@@ -307,6 +308,44 @@ impl Storage {
         drop(cursor);
         drop(block_txn);
 
+        info!("Checking block body db");
+        let body_txn = env.begin_ro_txn()?;
+        let mut cursor = body_txn.open_ro_cursor(block_body_db)?;
+
+        for (raw_key, raw_val) in cursor.iter() {
+            let body: BlockBody = lmdb_ext::deserialize(raw_val)?;
+            assert_eq!(
+                raw_key,
+                body.hash().as_ref(),
+                "found corrupt block body in database"
+            );
+        }
+        info!("block body db check complete");
+        drop(cursor);
+        drop(body_txn);
+
+        info!("Checking block_metadata_db");
+        let signature_txn = env.begin_ro_txn()?;
+        let mut cursor = signature_txn.open_ro_cursor(block_metadata_db)?;
+
+        for (raw_key, raw_val) in cursor.iter() {
+            let signature: BlockSignatures = lmdb_ext::deserialize(raw_val)?;
+            // Signature verification could be very slow process
+            // It iterates over every signature and verifies them.
+            match signature.verify() {
```

Comment:
I don't think we should be doing this. Signatures are verified before being stored in the storage. On the other hand, if it's being done only once at a startup maybe it's not a problem.

Reply:
Yeah it happens once during startup, but since we are checking  the integrity of the data, we can't rely on data being stored being in the same condition when we do startup. Its a slow process, but at this time I think its worth it.

Reply:
Yes, I agree. It's a good decision. We need to make sure that our data is correct :+1: